### PR TITLE
Rename `ListIsNullOrEmptyConverter` and `ListIsNotNullOrEmptyConverter`

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This PR implements the following name changes:
1. Rename `ListIsNotNullOrEmptyConverter` to `IsListNotNullOrEmptyConverter`
    - Moving the word `Is` to the front of the converter should improve discoverability by aligning with our other Converters that return a `bool`, e.g. `IsInRangeConverter`, `IsStringNullOrEmptyConverter` etc
2. Rename `ListIsNullOrEmptyConverter` to `IsListNullOrEmptyConverter`
    - Moving the word `Is` to the front of the converter should improve discoverability by aligning with our other Converters that return a `bool`, e.g. `IsInRangeConverter`, `IsStringNullOrEmptyConverter` etc

 ### Linked Issues ###
 

 - Fixes https://github.com/CommunityToolkit/Maui/pull/278#issuecomment-1031761255

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls/XYZ 

 ### Additional information ###

### CODE REVIEW

File: samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" encoding="utf-8" ?>
```
Comment (bijington) [inline samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml:1]: I've become so used to seeing this in XAML files I've never considered checking whether it needs to be there or not. My google skills are not helping me here, are we content that the files are going retain the utf8 encoding without this? Does it matter?

File: src/CommunityToolkit.Maui.UnitTests/Converters/IsListNotNullOrEmptyConverter_Tests.cs [hunk 2]
```diff
@@ -16,21 +16,21 @@ public class ListIsNotNullOrEmptyConverter_Tests : BaseTest
...
-		var result = (bool)listIsNotNullOrEmptyConverter.Convert(value, typeof(ListIsNotNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
+		var result = (bool)listIsNotNullOrEmptyConverter.Convert(value, typeof(IsListNotNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
 
-		Assert.Equal(result, expectedResult);
+		Assert.Equal(expectedResult, result);
 	}
 
 	[Theory]
 	[InlineData(0)]
 	public void InValidConverterValuesThrowArgumenException(object value)
```
Comment (bijington) [inline src/CommunityToolkit.Maui.UnitTests/Converters/IsListNotNullOrEmptyConverter_Tests.cs:30]: such a minor point but

```suggestion
	public void InvalidConverterValuesThrowArgumenException(object value)
```

File: src/CommunityToolkit.Maui.UnitTests/Converters/IsListNullOrEmptyConverter_Tests.cs [hunk 3]
```diff
@@ -16,21 +16,21 @@ public class ListIsNullOrEmptyConverter_Tests : BaseTest
...
-		var result = (bool)listIstNullOrEmptyConverter.Convert(value, typeof(ListIsNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
+		var result = (bool)listIstNullOrEmptyConverter.Convert(value, typeof(IsListNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
 
-		Assert.Equal(result, expectedResult);
+		Assert.Equal(expectedResult, result);
 	}
 
 	[Theory]
 	[InlineData(0)]
 	public void InValidConverterValuesThrowArgumenException(object value)
```
Comment (bijington) [inline src/CommunityToolkit.Maui.UnitTests/Converters/IsListNullOrEmptyConverter_Tests.cs:30]: Such a minor point but

```suggestion
	public void InvalidConverterValuesThrowArgumenException(object value)
```

File: samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml (hunk 1 above)
Comment (bijington) [reply samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml:1]: I don't really know if there would be implications here, I guess I have never questioned it's existence before 😄.

I'm all for consistency. Let's keep it out for now.

File: src/CommunityToolkit.Maui.UnitTests/Converters/IsListNullOrEmptyConverter_Tests.cs [hunk 4]
```diff
@@ -16,21 +16,21 @@ public class ListIsNullOrEmptyConverter_Tests : BaseTest
...
+		var result = (bool)listIstNullOrEmptyConverter.Convert(value, typeof(IsListNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
 
-		Assert.Equal(result, expectedResult);
+		Assert.Equal(expectedResult, result);
 	}
 
 	[Theory]
 	[InlineData(0)]
-	public void InValidConverterValuesThrowArgumenException(object value)
+	public void InvalidConverterValuesThrowArgumenException(object value)
```
Comment (bijington) [inline src/CommunityToolkit.Maui.UnitTests/Converters/IsListNullOrEmptyConverter_Tests.cs:30]: Another minor point but:

```suggestion
	public void InvalidConverterValuesThrowArgumentException(object value)
```

File: src/CommunityToolkit.Maui.UnitTests/Converters/IsListNotNullOrEmptyConverter_Tests.cs [hunk 5]
```diff
@@ -16,21 +16,21 @@ public class ListIsNotNullOrEmptyConverter_Tests : BaseTest
...
+		var result = (bool)listIsNotNullOrEmptyConverter.Convert(value, typeof(IsListNotNullOrEmptyConverter), null, CultureInfo.CurrentCulture);
 
-		Assert.Equal(result, expectedResult);
+		Assert.Equal(expectedResult, result);
 	}
 
 	[Theory]
 	[InlineData(0)]
-	public void InValidConverterValuesThrowArgumenException(object value)
+	public void InvalidConverterValuesThrowArgumenException(object value)
```
Comment (bijington) [inline src/CommunityToolkit.Maui.UnitTests/Converters/IsListNotNullOrEmptyConverter_Tests.cs:30]: Another minor point:

```suggestion
	public void InvalidConverterValuesThrowArgumentException(object value)
```

**Comment by brminnick** [comment]
Thanks Shaun! I'm the worst at typos 😅

**Comment by bijington** [comment]
> Thanks Shaun! I'm the worst at typos 😅

You and me both! 😄

**Pull request merged**